service/resource/s3objectv2/resource.go

```diff
@@ -97,45 +99,38 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	return &framework.Patch{}, nil
+	return framework.NewPatch(), nil
```

Comment:
Little bit of boyscouting

Reply:
I think it should be valid to just pass nil in case there is nothing to do. We should always allow zero values to be passed as a default for no action to be taken. /cc @kopiczko

Reply:
I think supporting nil values is good and I think it makes the framework easier to use. But I thought using `framework.NewPatch` is what we were doing currently.

Happy to change this later but would be good to get Pawel's input on it.

Reply:
This is good now. Just saying. This could be improved in general.